### DISCUSSION TITLE
Is it possible to change the commands at the time of deployment?

### DISCUSSION THREAD

**Discussion by koala-dev-123** [argoproj/argo-cd#8206] [Q&A]
Is it possible to change the commands at the time of deployment?
When deploying with helm, I want to deploy using another command instead of "helm install".
Another command is customization. Examples are "helm installb" or "helm ins".

**Comment by alexmt** [comment]
Argo CD does not use `helm install` to apply changes. Instead if uses `helm template` and apply produced resources using `kubectl apply. You can customize the process using [sync options](https://argo-cd.readthedocs.io/en/stable/user-guide/sync-options/) and [sync hooks](https://argo-cd.readthedocs.io/en/stable/user-guide/resource_hooks/)

**Reply by koala-dev-123** [reply]
Thank.I'm glad to receive a reply.

Is the flow "Repository change-> helm template-> kubectl apply-> Deployment completed"?
Where can I find it? Please tell me the URL.

 I want to use customized deploy command.Can't it?
"Repository change-> helm installb(customized deploy command)->Deployment completed"

**Reply by meln5674** [reply]
You can see how the command is built up here: https://github.com/argoproj/argo-cd/blob/master/reposerver/repository/repository.go . You'll find that there's no way to customize it within argocd itself. If you really need this functionality, my suggestion would be to build a custom image where the "helm" on the $PATH is actually a script of your own design which calls the "real" helm the way you want it to with the rest of the arguments.
